### PR TITLE
refactor Date model

### PR DESCRIPTION
**Pull request by jackmacaa**

### CODE REVIEW

File: app/models/Date.php [hunk 1]
```diff
@@ -1,59 +1,40 @@
...
-            $this->diff = abs($this->date2 - $this->date1);
-
-            $years = floor($this->diff / (365 * 60 * 60 * 24));
-            $months = floor(($this->diff - $years * 365 * 60 * 60 * 24) / (30 * 60 * 60 * 24));
-
-            return floor(($this->diff + $years * 365 * 60 * 60 * 24 + $months * 30 * 60 * 60 * 24) / (60 * 60 * 24));
-        }
+        protected $date1;
+        protected $date2;
+        protected $diff;
```
Comment (toddhainsworth) [inline app/models/Date.php]: These 3 need a type added to them (DateTime?)

File: app/models/Date.php [hunk 2]
```diff
@@ -1,59 +1,40 @@
...
 
         public function weekDays($data):int{
```
Comment (toddhainsworth) [inline app/models/Date.php]: `$data` needs a type added to it

File: app/views/dates/difference.php [hunk 3]
```diff
@@ -5,7 +5,7 @@
     <a href="<?= URLROOT; ?>/dates">Back</a>
     <div class="card card-body bg-light mt-5">
         <h3><strong><?= $data[0]; ?></strong>
-            <?php if($data['returnformat'] == 'days' || $data['returnformat'] == 'weeks' || $data['returnformat'] == 'years') : ?>
+            <?php if($data['returnformat'] == 'days' || $data['returnformat'] == 'weeks' || $data['returnformat'] == 'years' || $data['returnformat'] == 'weekdays') : ?>
```
Comment (toddhainsworth) [inline app/views/dates/difference.php]: This might be neater:
`if (in_array($data['returnformat'], ['days', 'weeks', 'years', 'weekdays']))`

**Review by toddhainsworth — CHANGES_REQUESTED** [review]

File: app/views/dates/difference.php (hunk 3 above)
Comment (jackmacaa) [reply app/views/dates/difference.php]: ooo nice

**Review by toddhainsworth — APPROVED** [review]
Nice, I like the way you've made the date model more polymorphic!

**Pull request merged**